src/interface/svd.jl

```diff
@@ -104,8 +104,7 @@ for f in (:svd_full!, :svd_compact!, :svd_vals!)
     end
 end
 
-function select_algorithm(::typeof(svd_trunc!), ::Type{A}, alg; trunc=nothing,
-                          kwargs...) where {A<:YALAPACK.BlasMat}
+function select_algorithm(::typeof(svd_trunc!), A, alg; trunc=nothing, kwargs...)
     alg_svd = select_algorithm(svd_compact!, A, alg; kwargs...)
```

Comment:
Though this is not strictly part of this PR, I am a bit suspicious aobut hte `svd_trunc!` and `eig_trunc!` rules. What If I call `svd_trunc!` and I already specify a `TruncatedAlgorithm` as `alg`. This will then be passed on to the `svd_compact!` select algorithm call, and then be wrapped once more in a new `TruncatedAlgorithm`?

Reply:
Good point, I can look into how that case is handled. Maybe we can just have a branch here like:
```julia
function select_algorithm(::typeof(svd_trunc!), A, alg; trunc=nothing, kwargs...)
    if alg isa TruncatedAlgorithm
        # Maybe assert that `isnothing(trunc)`
        return alg
    else
        alg_svd = select_algorithm(svd_compact!, A, alg; kwargs...)
        return TruncatedAlgorithm(alg_svd, select_truncation(trunc))
    end
end
```

Reply:
I've addressed this using that code pattern and added corresponding tests.